pypit/core/pydl.py

```diff
@@ -204,27 +204,27 @@ class bspline(object):
     Attributes
     ----------
     breakpoints
-        To be documented.
+        Breakpoints for bspline, spacing for these breakpoints are determinated by keywords inputs;
     nord
-        To be documented.
+        Order of bspline; [default=4]
     npoly
-        To be documented.
+        Polynomial order to fit over 2nd variable (when specified as x2): [default=1]
     mask
-        To be documented.
+        Output mask, set =1 for good points, =0 for bad points;
     coeff
-        To be documented.
+        Output coefficient of the bspline;
     icoeff
-        To be documented.
+        Cholesky band matrix used to solve for the bspline coefficients;
     xmin
-        To be documented.
+        Normalization minimum for x2; [default max(xdata)]
     xmax
-        To be documented.
+        Normalization maximum for x2; [default min(xdata)]
     funcname
-        To be documented.
+        Function for the second variable; [default 'legendre']
     """
 
     def __init__(self, x, nord=4, npoly=1, bkpt=None, fullbkpt = None, bkspread=1.0,
```

Comment:
I'd prefer if you broke this out into an `__init__` and `from_dict` methods.  Like this:

```
    def __init__(self, x, nord=4, npoly=1, bkpt=None, fullbkpt=None, mask=None, coeff=None,
                 icoeff=None, xmin=None, xmax=None, funcname=None, bkspread=1.0, verbose=False,
                 **kwargs):

        ...

    @classmethod
    def from_dict(cls, x, d):
        return cls(**d)

        # or

        return cls(x, nord=d['nord'], npoly=d['npoly'], bkpt=d['bkpt'], mask=d['mask'],
                   coeff=d['coeff'], icoeff=d['icoeff'], xmin=d['xmin'], xmax=d['xmax'],
                   funcname=d['funcname'])
```

I'd also remove the kwargs and add things to the argument list and dictionary.

Reply:
@kbwestfall, I don't see the point of an separate from_dict method, since we would just be instantiating an empty bspline object, and then reinstantiating it from a dict. The to_dict is a method, but the from_dict, needs to be part of the __init__.

Reply:
@kbwestfall regarding the kwargs, this definitely should be fixed, but not in this pull request. The kwargs were inherited from the initial pydl implementation written by Ben Weaver and have not been changed yet. 

Joe

Reply:
classmethods are the same as instantiation.  I.e., if you break this out like I suggested, you have two ways of instantiating the object, one from a dictionary and one from explicitly giving the keyword arguments.

After following what I suggest, you would not do:
```
b = bspline(x)
b.from_dict(x,d)
```
You would just say:
```
b = bspline.from_dict(x,d)
```

There's no reason you can't do it the way you have it, it's just cleaner if you do it the way I suggest.

Reply:
Okay @kbwestfall , I agree your way is cleaner and more transparent. We will make it so. 

Joe